libraries/AP_Compass/AP_Compass_Calibration.cpp

```diff
@@ -33,20 +35,22 @@ Compass::compass_cal_update()
     if (is_calibrating()) {
         _cal_has_run = true;
         return;
-    } else if (_cal_has_run && auto_reboot()) {
+    } else if (_cal_has_run && _auto_reboot()) {
         hal.scheduler->delay(1000);
         hal.scheduler->reboot(false);
     }
 }
 
 bool
-Compass::start_calibration(uint8_t i, bool retry, bool autosave, float delay, bool autoreboot)
+Compass::_start_calibration(uint8_t i, bool retry, float delay)
 {
     if (!healthy(i)) {
         return false;
     }
-    memset(_reports_sent,0,sizeof(_reports_sent));
-    if (!is_calibrating() && delay > 0.5f) {
+    if (!use_for_yaw(i)) {
+        return false;
```

Comment:
It would only try to calibrate a disabled compass if it was specifically asked to do it. And I don't see why not. If it is asked to calibrate compass 1 it should try to calibrate it no matter what.

If a GCS doesn't provide a way for the user to choose what he wants to do then it should use value 0 which would have the special value "calibrate all flight usable compasses", excluding the disabled compasses. That might not be clear in the MAVLink message but Jon confirmed that was the intended behavior.